src/net_socket.c

```diff
@@ -288,10 +288,40 @@ int setup_vpn_in_socket(const sockaddr_t *sa) {
 	if(udp_rcvbuf && setsockopt(nfd, SOL_SOCKET, SO_RCVBUF, (void *)&udp_rcvbuf, sizeof(udp_rcvbuf))) {
 		logger(DEBUG_ALWAYS, LOG_WARNING, "Can't set UDP SO_RCVBUF to %i: %s", udp_rcvbuf, sockstrerror(sockerrno));
 	}
+	{
+		// The system may cap the requested buffer size.
+		// Read back the value and check if it is now as requested.
+		int udp_rcvbuf_actual = -1;
+		socklen_t optlen = sizeof(udp_rcvbuf_actual);
+		if (getsockopt(nfd, SOL_SOCKET, SO_RCVBUF, (void *)&udp_rcvbuf_actual, &optlen)) {
+			logger(DEBUG_ALWAYS, LOG_WARNING, "Can't read back UDP SO_RCVBUF: %s", sockstrerror(sockerrno));
+		} else if(optlen != sizeof(udp_rcvbuf_actual)) {
+			logger(DEBUG_ALWAYS, LOG_WARNING, "Can't read back UDP SO_RCVBUF: Unexpected returned optlen %i", optlen);
```

Comment:
`optlen` is a `socklen_t`, you cannot assume it is the same as an `int`. So `%i` is not safe to use, unless you explicitly cast `optlen` to `int`.

Reply:
Thanks, addressed as described on https://stackoverflow.com/questions/39417560/correct-format-specifiers-for-custom-integer-types-in-c-to-ensure-portability